### PR TITLE
Spark 3.3: Use ProcedureInput in AddFilesProcedure

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR migrates `AddFilesProcedure` to use `ProcedureInput`.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
cc @szehon-ho @amogh-jahagirdar @flyrain @singhpk234 @ajantha-bhat

**Review by singhpk234 — APPROVED** [review]
LGTM, Thanks @aokolnychyi !

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java [hunk 1]
```diff
@@ -167,6 +167,11 @@ public Identifier ident(ProcedureParameter param) {
     return catalogAndIdent.identifier();
   }
 
+  public Identifier ident(ProcedureParameter param, CatalogPlugin defaultCatalog) {
```
[the remark this answers is not in the document]
Comment (aokolnychyi) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java:169]: I think we can't use it as we need to validate the catalog in that method. However, I changed the code a bit to share more. @singhpk234, could you take another look?

[the remark this answers is not in the document]
Comment (singhpk234) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java:169]: Agree, by above, I meant that modifying this function to just get identifier from `ident(ProcedureParameter param, CatalogPlugin defaultCatalog)` and keep the assertion as it is, hence specifically line 158.

--- 

> However, I changed the code a bit to share more. @singhpk234, could you take another look?

+1 This looks great !

**Review by singhpk234 — APPROVED** [review]

**Review by flyrain — APPROVED** [review]
+1. Thanks @aokolnychyi! Do we need the `ProcedureInput` change in Spark 3.2?

**Comment by aokolnychyi** [comment]
>  Do we need the ProcedureInput change in Spark 3.2?

@flyrain, I was debating that. Probably we could do that for consistency? What are your thoughts?

**Comment by flyrain** [comment]
Yeah, my only concern is that, any new change to procedure will be hard to apply to 3.2 if we don't do that.

**Comment by aokolnychyi** [comment]
Agreed, will cherry-pick in a bit.

**Comment by aokolnychyi** [comment]
Thanks for reviewing, @singhpk234 and @flyrain!

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (szehon-ho) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java:169]: I am probably a bit late to comment, but I wanted to note, I think 'defaultCatalog' is a bit confusing here, as ProcedureInput already comes with 'catalog'.  To me, its more of a catalog override, isnt it?

[the remark this answers is not in the document]
Comment (aokolnychyi) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java:169]: Well, it kind of overrides the default catalog to use if the identifier does not reference a catalog explicitly. It is not necessarily the catalog that will be used. It matches the arg name in `Spark3Util`.

**Pull request merged**